README.md
# website_evergreen
Evergreen Sprout website


about/history.md
---
title: History of The Sprout Fund
---
<section class="pt-1">
  <div class="container text-center">
    <div class="row">
      <div class="col-md-10 offset-md-1 col-lg-8 offset-lg-2">
        <h1>History</h1>
        <p class="lead" style="margin-bottom: 2em;">For nearly two decades, Sprout’s partnerships with regional and national foundations helped power Pittsburgh’s grassroots response to a wide range of social, environmental, and economic challenges.</p>
      </div>
    </div>
  </div>
  <div class="container">
    <div class="row">
      <div class="col-12 text-center">
        <h2 style="margin-bottom: 1.181818181818185em;">Beginnings &amp; Early Success</h2>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2001</h4>
          <p>Created the <b>Seed Award</b> as mechanism for young people to support innovative ideas to make Pittsburgh a better place to live, work, and play. Supported initially by the Richard King Mellon Foundation and Heinz Endowments.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2003</h4>
          <p>Launched Public Art <b>community mural</b> program to transform the visual landscape and bring communities together around shared goals. Longstanding program supporters were PNC Foundation and Laurel Foundation.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2003</h4>
          <p>Hosted first <b>Hothouse</b> event, our “live annual report” to the community and annual fundraiser until 2010. It regularly earned “best party” accolades and drew crowds to recent redevelopments.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2004</h4>
          <p>Partnered with Club Café, The Andy Warhol Museum, and the Brew House Association to activate Pittsburgh in winter through a 3-month series of <b>art, music, and performances</b>.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2006</h4>
          <p>Issued first RFPs on community-generated topics through <b>Engage Pittsburgh</b> after the Idea Round Up. Lead support provided by Claude Worthington Benedum Foundation.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2006</h4>
          <p>Commissioned local artists to design <b>Giant Inflatables</b> that were erected at cultural sites throughout the city to raise awareness about the Pittsburgh Roars marketing campaign.</p>
        </div>
      </div>
      <div class="col-12 text-center">
        <h2 style="margin-bottom: 1.181818181818185em;">Growth &amp; Expansion</h2>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2008</h4>
          <p>Stewarded $1 million in Community Connections grants to help citizens take part in regional celebration of <b>Pittsburgh’s 250</b>th anniversary. Worked on behalf of Allegheny Conference on Community Development.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2009</h4>
          <p>Launched <b>Spark </b>program in partnership with The Grable Foundation to forge collaborations between early learning educators and digital media technologists.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2010</h4>
          <p>Partnered with The Pittsburgh Foundation to support local efforts to enhance <b>biodiversity</b> when Pittsburgh was North American host city for UN World Environment Day.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2011</h4>
          <p>Celebrated 10th anniversary of organization with <b><i>TENACITY</i></b>, a one-night-only performance retrospective from Sprout’s first 10 years of investments in the arts.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2012</h4>
          <p>Formalized the learning innovation network now called <b>Remake Learning</b> and helped it grow from a hundred people and a few dozen organizations to thousands of teachers, techies, and civic leaders from 500+ regional orgs.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2012</h4>
          <p>Issued first annual batch of $1,000 innovation grants that would come to be known as <b>Grand Ideas</b> as the conclusion of the year-long Social Innovation Exchange.</p>
        </div>
      </div>
      <div class="col-12 text-center">
        <h2 style="margin-bottom: 1.181818181818185em;">Regional &amp; National Leadership</h2>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2013</h4>
          <p>Secured national funding from <b>MacArthur Foundation</b> to establish a Hive and support Connected Learning activities for teens that were interest-driven, peer-supported, and production-centered.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2014</h4>
          <p>Hosted an <b>international summit in Silicon Valley</b> to focus the attention of educators and employers on the potential of open digital badges—on behalf of MacArthur Foundation and in partnership with Digital Promise.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2014</h4>
          <p>Established <b>The Digital Corps</b> to deploy trained-mentors into afterschool community centers to help improve digital literacy skills. Supported by The Grable Foundation and part of the Mozilla global community.</p>
        </div>
      </div>
      <div class="col-md-4">
        <div class="feature">
          <h4>2014</h4>

search/search.js
/* global instantsearch */

window.addEventListener('load', function() {
  var search = instantsearch({
    appId: 'NWCVVYDLQV',
    apiKey: '054fce7f83acc9d7d52aa88bcb8f997e',
    indexName: 'sproutfund_grants',
    routing: true
  });

  search.addWidget(
    instantsearch.widgets.searchBox({
      container: '#q',
      placeholder: 'Search projects & programs',
      magnifier: false,
      poweredBy: false,
      wrapInput: false
    })
  );

  search.addWidget(
    instantsearch.widgets.stats({
      container: '#stats'
    })
  );

  var modalTemplate =
    '<div class=""'

  var hitTemplate =
    '<div class="ais-hits--item--header">' +
      '{{#featured}}<a title="{{name}}, a {{superprogram}} Project Highlight" target="_blank" href="{{url}}">{{/featured}}' +
      '<h4>{{{_highlightResult.name.value}}}</h4>' +
      '{{#featured}}<span><i class="icon icon--xs stack-plus-circled"></i></span></a>{{/featured}}' +
    '</div>' +
    '<p>{{{_highlightResult.description.value}}}<br/><span class="type--fine-print">{{{_highlightResult.addon.value}}}</span>';

  var noResultsTemplate = '<div class="text-center">' +
    'No projects found{{#query}} matching <strong>{{query}}</strong>{{/query}}.' +
    ' <a class="btn btn--sm" href="."><span class="btn__text">Start Over</span></a>'
    '</div>';

  var checkboxRefinementListTemplate =
    '<div class="input-checkbox">' +
      '<input type="checkbox" class="ais-refinement-list--checkbox" value="{{label}}" {{#isRefined}}checked{{/isRefined}} />' +
      '<label for=""></label>' +
    '</div>' +
    '<span><span class="ais-refinement-list--label">{{label}}</span><span class="label facet-count">{{count}}</span></span>';

  var radioInnerRefinementListTemplate =
    '<div class="input-radio input-radio--innerlabel">' +
      '<input type="radio" class="ais-refinement-list--radio" value="{{label}}" {{#isRefined}}checked{{/isRefined}} />' +
      '<label for=""><span class="ais-refinement-list--label">{{label}}</span></label>' +
    '</div>';

  var toggleItemTemplate =
    '<div class="d-flex">' +
      '<div class="pr-2 pt-1"><span class="ais-toggle--label {{^isRefined}}active{{/isRefined}}">Show All</span></div>' +
      '<div class="align-self-center">' +
        '<div class="input-checkbox input-checkbox--switch">' +
          '<input type="checkbox" {{#isRefined}}checked{{/isRefined}}/>' +
          '<label for=""></label>' +
        '</div>' +
      '</div>' +
      '<div class="pl-2 pt-1"><span class="ais-toggle--label {{#isRefined}}active{{/isRefined}}">Only Highlights</span></div>' +
    '</div>';

  search.addWidget(
      instantsearch.widgets.hits({
        container: '#hits',
        templates: {
          empty: noResultsTemplate,
          item: hitTemplate
        }
      })
    );

  search.addWidget(
    instantsearch.widgets.rangeInput({
      container: '#filter_year',
      attributeName: 'year',
      cssClasses: {
        submit   : 'btn btn--secondary'
      },
      labels: {
        separator: ' to ',
        submit: 'Filter'
      },
      templates: {
        header: '<h5>Year</h5>'
      },
      autoHideContainer: true
    })
  );

  search.addWidget(
    instantsearch.widgets.refinementList({
      container: '#filter_superprogram',
      attributeName: 'superprogram',
      operator: 'or',
      limit: 10,
      sortBy: ['name:asc'],
      templates: {
        item: radioInnerRefinementListTemplate,
        header: '<h5>Category</h5>'
      },
      autoHideContainer: true
    })
  );

  search.addWidget(
    instantsearch.widgets.refinementList({
      container: '#filter_program',
      attributeName: 'program',
      operator: 'or',
      limit: 10,
      sortBy: ['count:desc'],
      showMore: {
        templates: {
          active: '<a class="ais-show-more ais-show-more__active" title="show less">less<i class="icon icon--xs stack-up-dir"></i></a>',
          inactive: '<a class="ais-show-more ais-show-more__inactive" title="show more">more<i class="icon icon--xs stack-down-dir"></i></a>'
        }
      },
      templates: {
        item: checkboxRefinementListTemplate,
        header: '<h5>Program</h5>'
      },
      autoHideContainer: true
    })
  );

  search.addWidget(
    instantsearch.widgets.refinementList({
      container: '#filter_type',
      attributeName: 'type',
      operator: 'or',
      sortBy: ['count:desc'],
      templates: {
        item: checkboxRefinementListTemplate,
        header: '<h5>Grant Type</h5>'
      },
      autoHideContainer: true
    })
  );

  search.addWidget(
    instantsearch.widgets.numericRefinementList({
      container: '#filter_amount',
      attributeName: 'amount',
      options: [
        {name: 'Any'},
        {              end:  1499, name: 'less than $1,500'},
        {start:  1500, end:  5000, name: '$1,500 – $5,000'},
        {start:  5001, end: 10000, name: '$5,001 – $10,000'},
        {start: 10001, end: 25000, name: '$10,001 – $25,000'},
        {start: 25001,             name: 'greater than $25,000'}
      ],
      templates: {
        item: radioInnerRefinementListTemplate,
        header: '<h5>Amount</h5>'
      },
      autoHideContainer: false
    })
  );

  search.addWidget(
    instantsearch.widgets.toggle({
      container: '#filter_featured',
      attributeName: 'featured',
      label: 'Show Only Project Highlights',
      values: {
        on: true,
      },
      templates: {
        item: toggleItemTemplate,
        header: ''
      }
    })
  );

  search.addWidget(
    instantsearch.widgets.clearAll({
      container: '#clear-all',
      templates: {
        link: 'Reset All Filters'
      },
      cssClasses: {
        root: 'btn btn-block btn-default'
      },
      autoHideContainer: true
    })
  );

  // custom `renderFn` to render the custom Pagination widget
  function renderFn(PaginationRenderingOptions, isFirstRendering) {
    if (isFirstRendering) {
      PaginationRenderingOptions.widgetParams.containerNode.html('<div class="pagination"><a class="pagination__prev" href="#" title="Previous Page">«</a><ol></ol><a class="pagination__next" href="#" title="Next Page">»</a></div>');
    }

    // remove event listeners before replacing markup
    PaginationRenderingOptions.widgetParams.containerNode
      .find('a[data-page]')
      .each(function() { $(this).off('click'); });

    var pages = PaginationRenderingOptions.pages
      .map(function(page) {
        if (PaginationRenderingOptions.currentRefinement == parseInt(page)) {
          return '<li class="pagination__current">' + (parseInt(page) + 1) + '</li>';
        } else {
          return '<li><a href="' + PaginationRenderingOptions.createURL(page) + '" data-page="' + page + '">' +
            (parseInt(page) + 1) + '</a></li>';
        }
      });

    var previousPageNum = parseInt(PaginationRenderingOptions.currentRefinement) - 1;
    if (PaginationRenderingOptions.isFirstPage) {
      PaginationRenderingOptions.widgetParams.containerNode
        .find('.pagination__prev')
        .removeAttr("href");
    } else {
      PaginationRenderingOptions.widgetParams.containerNode
        .find('.pagination__prev')
        .attr("href", PaginationRenderingOptions.createURL(previousPageNum) );
    }

    var nextPageNum = parseInt(PaginationRenderingOptions.currentRefinement) + 1;
    if (PaginationRenderingOptions.isLastPage) {
      PaginationRenderingOptions.widgetParams.containerNode
        .find('.pagination__next')
        .removeAttr("href");
    } else {
      PaginationRenderingOptions.widgetParams.containerNode
        .find('.pagination__next')
        .attr("href", PaginationRenderingOptions.createURL(nextPageNum) );
    }

    PaginationRenderingOptions.widgetParams.containerNode
      .find('ol')
      .html(pages);

    PaginationRenderingOptions.widgetParams.containerNode
      .find('a[data-page]')
      .each(function() {
        $(this).on('click', function(event) {
          event.preventDefault();
          PaginationRenderingOptions.refine($(this).data('page'));
        });
      });

    if (PaginationRenderingOptions.nbPages === 0 || PaginationRenderingOptions.nbPages === 1) {
      PaginationRenderingOptions.widgetParams.containerNode.attr("style", "display: none;");
    } else {
      PaginationRenderingOptions.widgetParams.containerNode.attr("style", "display: block;");
    }

  }

  // connect `renderFn` to Pagination logic
  var customPagination = instantsearch.connectors.connectPagination(renderFn);

  // mount custom Pagination widget on the page
  search.addWidget(
    customPagination({
      containerNode: $('#custom-pagination-container'),
      padding: 3,
    })
  );

  search.start();

});
